src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machines = new BulldozerProducer().get();
+        }
+        if (type == Truck.class) {
+            machines = new TruckProducer().get();
+        }
+        if (type == Excavator.class) {
+            machines = new ExcavatorProducer().get();
+        }
+        return new ArrayList<>(machines);
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        int size = machines.size();
```

Comment:
Maybe it would be better to relocate machine.size() within the for loop, without creating new variable(int size)?

Reply:
thanks, fix